src/org/labkey/targetedms/view/chromatogramLibraryDownload.jsp

```diff
@@ -28,51 +28,98 @@
 <%@ page import="java.nio.file.Files" %>
 <%@ page import="java.nio.file.Path" %>
 <%@ page import="java.text.DecimalFormat" %>
+<%@ page import="org.labkey.api.security.permissions.InsertPermission" %>
+<%@ page import="org.labkey.api.util.StringUtilsLabKey" %>
 <%@ page extends="org.labkey.api.jsp.JspBase" %>
 <%
     Container c = getContainer();
     User user = getUser();
     TargetedMSService.FolderType folderType = TargetedMSManager.getFolderType(c);
 
-    long peptideGroupCount = TargetedMSController.getNumRepresentativeProteins(user, c);
-    long peptideCount = TargetedMSController.getNumRepresentativePeptides(c);
-    long transitionCount = TargetedMSController.getNumRankedTransitions(c);
-    DecimalFormat format = new DecimalFormat("#,###");
+    long conflictCount = ConflictResultsManager.getConflictCount(user, c);
+    boolean hasConflicts = conflictCount != 0;
     int currentRevision = ChromatogramLibraryUtils.getCurrentRevision(c, user);
-    Path archiveFile = ChromatogramLibraryUtils.getChromLibFile(c, currentRevision);
+    int libRevision = hasConflicts ? ChromatogramLibraryUtils.getLastStableLibRevision(c, user) : currentRevision;
+
+    TargetedMSController.ChromLibAnalyteCounts chromLibAnalyteCounts = TargetedMSController.getChromLibAnalyteCounts(user, c, libRevision);
```

Comment:
Consider returning a ChromLibAnalyteCounts with zeros instead of a null to make this code a little cleaner. Could add an exists() method if other cases need to distinguish between null and a library with no data